### PR TITLE
deepcopy fix steps

### PR DESCRIPTION
**Pull request by lut777**
To avoid directly modification to client-go data structures in mem, Deep-Copy is required.
As title, try to fix it in steps.

### CODE REVIEW

File: pkg/controller/pod.go [hunk 1]
```diff
@@ -618,13 +619,14 @@ func (c *Controller) handleUpdatePodSecurity(key string) error {
 		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
 		return nil
 	}
-	pod, err := c.podsLister.Pods(namespace).Get(name)
+	oripod, err := c.podsLister.Pods(namespace).Get(name)
```
Comment (hongzhen-ma) [inline pkg/controller/pod.go:622]: the variable pod here is not modified in context

File: pkg/controller/subnet.go [hunk 2]
```diff
@@ -429,13 +429,14 @@ func (c Controller) patchSubnetStatus(subnet *kubeovnv1.Subnet, reason string, e
 
 func (c *Controller) handleAddOrUpdateSubnet(key string) error {
 	var err error
-	subnet, err := c.subnetsLister.Get(key)
+	orisubnet, err := c.subnetsLister.Get(key)
```
Comment (hongzhen-ma) [inline pkg/controller/subnet.go:432]: here is used for finalizer, the actual used variable is downside, so it is ok not to modify here.

**Review by hongzhen-ma — CHANGES_REQUESTED** [review]

File: pkg/controller/init.go [hunk 3]
```diff
@@ -90,11 +90,12 @@ func (c *Controller) InitDefaultVpc() error {
 
 // InitDefaultLogicalSwitch init the default logical switch for ovn network
 func (c *Controller) initDefaultLogicalSwitch() error {
-	subnet, err := c.config.KubeOvnClient.KubeovnV1().Subnets().Get(context.Background(), c.config.DefaultLogicalSwitch, metav1.GetOptions{})
+	orisubnet, err := c.config.KubeOvnClient.KubeovnV1().Subnets().Get(context.Background(), c.config.DefaultLogicalSwitch, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (lut777) [reply pkg/controller/init.go:93]: right. modified.

File: pkg/controller/init.go [hunk 4]
```diff
@@ -136,11 +137,12 @@ func (c *Controller) initDefaultLogicalSwitch() error {
 
 // InitNodeSwitch init node switch to connect host and pod
 func (c *Controller) initNodeSwitch() error {
-	subnet, err := c.config.KubeOvnClient.KubeovnV1().Subnets().Get(context.Background(), c.config.NodeSwitch, metav1.GetOptions{})
+	orisubnet, err := c.config.KubeOvnClient.KubeovnV1().Subnets().Get(context.Background(), c.config.NodeSwitch, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (lut777) [reply pkg/controller/init.go:140]: modified

File: pkg/controller/node.go [hunk 5]
```diff
@@ -503,7 +505,7 @@ func (c *Controller) handleUpdateNode(key string) error {
 
 func (c *Controller) createOrUpdateCrdIPs(key, ip, mac string) error {
 	v4IP, v6IP := util.SplitStringIP(ip)
-	ipCr, err := c.config.KubeOvnClient.KubeovnV1().IPs().Get(context.Background(), fmt.Sprintf("node-%s", key), metav1.GetOptions{})
+	oriipCr, err := c.config.KubeOvnClient.KubeovnV1().IPs().Get(context.Background(), fmt.Sprintf("node-%s", key), metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (lut777) [reply pkg/controller/node.go:508]: modified

File: pkg/controller/pod.go (hunk 1 above)
Comment (lut777) [reply pkg/controller/pod.go:622]: modified

File: pkg/controller/subnet.go (hunk 2 above)
Comment (lut777) [reply pkg/controller/subnet.go]: right. modified

File: pkg/controller/init.go [hunk 6]
```diff
@@ -54,17 +54,17 @@ func (c *Controller) InitOVN() error {
 }
 
 func (c *Controller) InitDefaultVpc() error {
-	vpc, err := c.vpcsLister.Get(util.DefaultVpc)
+	orivpc, err := c.vpcsLister.Get(util.DefaultVpc)
```
[the remark this answers is not in the document]
Comment (lut777) [reply pkg/controller/init.go:57]: true, but lets keep it just in case.

**Review by hongzhen-ma — APPROVED** [review]

**Pull request merged**